aspnetcore/blazor/file-downloads.md

```diff
@@ -102,23 +100,31 @@ async function downloadFileFromStream(fileName, contentStreamReference) {
 function triggerFileDownload(fileName, url) {
   const anchorElement = document.createElement('a');
   anchorElement.href = url;
-
-  if (fileName) {
-    anchorElement.download = fileName;
-  }
-
+  anchorElement.download = fileName ?? '';
   anchorElement.click();
   anchorElement.remove();
 }
 ```
 
+`contentStreamReference.arrayBuffer()` loads the entire file into client memory. For file downloads over 250 MB, we recommend opening a new tab for each large file download by setting the anchor's [`target`](https://developer.mozilla.org/docs/Web/HTML/Element/a#attr-target) to `_blank` and [`referrerpolicy`](https://developer.mozilla.org/docs/Web/HTML/Element/a#attr-referrerpolicy) to `no-referrer`:
+
+```javascript
+anchorElement.target = '_blank';
+anchorElement.referrerpolicy = 'no-referrer';
+```
```

Comment:
```suggestion
`contentStreamReference.arrayBuffer()` loads the entire file into client memory. For file downloads over 250 MB, we recommend downloading the file from a URL instead:
```

```csharp
<button @onclick="DownloadFileFromURL">
    Download File From Stream
</button>

@code {
    private async Task DownloadFileFromURL()
    {
        var fileURL = "https://aka.ms/dotnet/7.0.1xx/daily/dotnet-sdk-win-x64.exe";
        var fileName = "dotnet-7-nightly";
        await JS.InvokeVoidAsync("triggerFileDownload", fileName, fileURL);
    }
}
```

(feel free to replace the download link with something else, this is just random)

Maybe worth including a blurb about "Please ensure security best practices are followed when providing a URL to download the file" (example text feel free to change as you see fit).

Reply:
Thanks @TanayParikh ... I'll get this set up in the topic on Tuesday and ping u back. Have a nice holiday weekend! 🌴

Reply:
Thanks, you too!